salt/fileclient.py

```diff
@@ -486,8 +486,11 @@ def get_url(self, url, dest, makedirs=False, saltenv='base',
                     'Path \'{0}\' is not absolute'.format(url_path)
                 )
             if dest is None:
-                with salt.utils.fopen(url_path, 'r') as fp_:
+                mode = 'rb' if six.PY3 else 'r'
+                with salt.utils.fopen(url_path, mode) as fp_:
                     data = fp_.read()
+                    if six.PY3:
+                        data = data.decode()
```

Comment:
The problem is it is currently exploding when I set this to __salt_system_encoding__, because that is 'us-ascii' in this case.